src/CommonITILValidation.php

```diff
@@ -1993,4 +1947,152 @@ public static function getAllValidationStatusArray()
+        $itil_id = $this->input[$itilobject_foreignkey] ?? $this->fields[$itilobject_foreignkey];
+        $itil_validations = $v->find(
+            [$itilobject_foreignkey => $itil_id,
+                ['NOT' => ['itils_validationsteps_id' => 0]],
+            ]
+        );
+        // find an itils_validationsteps related to the ticket and the validationstep
+        $itils_validationsteps_ids = array_column($itil_validations, 'itils_validationsteps_id');
+        $ivs = $itilobject_type::getValidationStepInstance();
+        $itils_validationsteps_id = $itils_validationsteps_ids[0] ?? null; // there can be only one itil_validationsteps_id related to the validationstep (or none)
+        // check if the itils_validationsteps_id is referencing the validationstep
+        $itils_validationstep_id_with_validationstep_id_exists =
+            !is_null($itils_validationsteps_id)
+            && $ivs->getFromDBByCrit(['id' => $itils_validationsteps_ids, 'validationsteps_id' => $validationsteps_id]);
+
+        if ($itils_validationstep_id_with_validationstep_id_exists) {
+            $itils_validationsteps_id = $ivs->fields['id'];
+            $this->input['itils_validationsteps_id'] = $itils_validationsteps_id;
+        } else {
+            // addValidationStep also update the current Validation
+            // load referenced ValidationStep
+            $vs = new ValidationStep();
+            if (!$vs->getFromDB($validationsteps_id)) {
+                Session::addMessageAfterRedirect('Failed to load approval step while adding approval step.');
+            };
+
+            // create ITIL_ValidationStep
+            $itil_class = static::getItilObjectItemType();
+            $itil_validationstep = $itil_class::getValidationStepInstance();
+            $itil_validationstep->add([
+                'validationsteps_id' => $validationsteps_id,
+                'minimal_required_validation_percent' => $vs->fields['minimal_required_validation_percent'],
+            ]);
+
+            $this->input['itils_validationsteps_id'] = $itil_validationstep->getID();
+        }
+
+        unset($this->input['_validationsteps_id']);
+    }
+
+    /**
+     * Delete, only if the itils_validationstep is not used anymore
+     *
+     * @param int $itils_validationsteps_id
+     * @return void
+     */
+    private function removeUnsedITILValidationStep(int $itils_validationsteps_id): void
+    {
+        /** @var \CommonITILObject $itil */
+        $itil = $this->getItem();
+        $validation = $itil::getValidationClassInstance();
+        $validations = $validation->find(['itils_validationsteps_id' => $itils_validationsteps_id]);
+        if (!empty($validations)) {
+            // itils_validation is still used, do not delete
+            return;
+        }
+
+        if (!$itil::getValidationStepInstance()->delete(['id' => $itils_validationsteps_id])) {
+            Session::addMessageAfterRedirect('Failed to delete unused approval step.');
```

Comment:
`addMessageAfterRedirect` defaults to an INFO type message. Failures should be at least WARNING level.

Reply:
It seems fairly generic and we have no real information about the failure here. If the `delete` method fails, there may be something else that is logging the cause. I think the current message is OK just to let the user know something went wrong.